packages/core/admin/admin/src/pages/SettingsPage/pages/Webhooks/EditView/components/Events/index.js

```diff
@@ -0,0 +1,316 @@
+          })}
+        </Loader>
+      )}
+      <StyledTable>{renderChildren({ isDraftAndPublish })}</StyledTable>
+    </Flex>
+  );
+};
+
+Root.propTypes = {
+  renderChildren: PropTypes.func.isRequired,
+};
+
+const Headers = ({ getHeaders = getCEHeaders, isDraftAndPublish = false }) => {
+  const { formatMessage } = useIntl();
+  const headers = getHeaders(isDraftAndPublish);
+
+  return (
+    <Thead>
+      <Tr>
+        <Td />
+        {headers.map((header) => {
+          if (header.id === 'app.utils.publish' || header.id === 'app.utils.unpublish') {
+            return (
+              <Td
+                key={header.id}
+                title={formatMessage({
+                  id: 'Settings.webhooks.event.publish-tooltip',
+                  defaultMessage: 'This event only exists for content with draft & publish enabled',
+                })}
+              >
+                <Typography variant="sigma" textColor="neutral600">
+                  {formatMessage(header)}
+                </Typography>
+              </Td>
+            );
+          }
+
+          return (
+            <Td key={header.id}>
+              <Typography variant="sigma" textColor="neutral600">
+                {formatMessage(header)}
+              </Typography>
+            </Td>
+          );
+        })}
+      </Tr>
+    </Thead>
+  );
+};
+
+Headers.defaultProps = {
+  getHeaders: getCEHeaders,
+  isDraftAndPublish: false,
+};
+
+Headers.propTypes = {
+  getHeaders: PropTypes.func,
+  isDraftAndPublish: PropTypes.bool,
+};
+
+const Body = ({ providedEvents, isDraftAndPublish }) => {
+  const events = providedEvents || getCEEvents(isDraftAndPublish);
+  const { values, handleChange: onChange } = useFormikContext();
+
+  const inputName = 'events';
+  const inputValue = values.events;
+  const disabledEvents = [];
+
+  const formattedValue = formatValue(inputValue);
+
+  const handleChange = ({ target: { name, value } }) => {
+    let set = new Set(inputValue);
+
+    if (value) {
+      set.add(name);
+    } else {
+      set.delete(name);
+    }
+    onChange({ target: { name: inputName, value: Array.from(set) } });
+  };
+
+  const handleChangeAll = ({ target: { name, value } }) => {
+    let set = new Set(inputValue);
+
+    if (value) {
+      events[name].forEach((event) => {
+        if (!disabledEvents.includes(event)) {
+          set.add(event);
+        }
+      });
+    } else {
+      events[name].forEach((event) => set.delete(event));
+    }
+    onChange({ target: { name: inputName, value: Array.from(set) } });
+  };
+
+  return (
+    <Tbody>
+      {Object.entries(events).map(([event, value]) => {
+        return (
+          <EventRow
+            disabledEvents={disabledEvents}
+            key={event}
+            name={event}
+            events={value}
+            inputValue={formattedValue[event]}
+            handleChange={handleChange}
+            handleChangeAll={handleChangeAll}
```

Comment:
Are these triggered when the checkbox is checked/ unchecked? In case it is: what do you think about a better name such as: `handleSelect` and `handleSelectAll`?